test/e2e/istio_test.go

```diff
@@ -0,0 +1,145 @@
+	handleSuiteTearDown()
+}
+
+// LIFECYCLE - Test
+
+func (suite *IstioTestSuite) SetupTest() {
+	t = suite.T()
+}
+
+func (suite *IstioTestSuite) AfterTest(suiteName, testName string) {
+	handleTestFailure()
+}
+
+func TestIstioSuite(t *testing.T) {
+	suite.Run(t, new(IstioTestSuite))
+}
+
+func (suite *IstioTestSuite) TestEnvoySidecar() {
+	// First deploy a Jaeger instance
+	jaegerInstanceName := "simplest"
+	jaegerInstance := createJaegerInstanceFromFile(jaegerInstanceName, "../../examples/simplest.yaml")
+	defer undeployJaegerInstance(jaegerInstance)
+	err := WaitForDeployment(t, fw.KubeClient, namespace, jaegerInstanceName, 1, retryInterval, timeout+(1*time.Minute))
+	require.NoError(t, err)
+
+	// Now deploy examples/business-application-injected-sidecar.yaml
+	businessAppCR := getBusinessAppCR(err)
+	defer os.Remove(businessAppCR.Name())
+	cmd := exec.Command("kubectl", "create", "--namespace", namespace, "--filename", businessAppCR.Name())
+	output, err := cmd.CombinedOutput()
+	if err != nil && !strings.Contains(string(output), "AlreadyExists") {
+		require.NoError(t, err, "Failed creating Jaeger instance with: [%s]\n", string(output))
+	}
+	const vertxDeploymentName = "myapp"
+	err = WaitForDeployment(t, fw.KubeClient, namespace, vertxDeploymentName, 1, retryInterval, timeout)
+	require.NoError(t, err, "Failed waiting for myapp deployment")
+
+	// Add a liveliness probe to create some traces
+	vertxPort := intstr.IntOrString{IntVal: 8080}
+	livelinessHandler := &corev1.HTTPGetAction{Path: "/", Port: vertxPort, Scheme: corev1.URISchemeHTTP}
+	handler := &corev1.Handler{HTTPGet: livelinessHandler}
+	livelinessProbe := &corev1.Probe{Handler: *handler, InitialDelaySeconds: 1, FailureThreshold: 3, PeriodSeconds: 10, SuccessThreshold: 1, TimeoutSeconds: 1}
+
+	err = waitForDeploymentAndUpdate(vertxDeploymentName, vertxDeploymentName, func(container *corev1.Container) {
+		container.LivenessProbe = livelinessProbe
+	})
+	require.NoError(t, err)
+
+	exists := testContainerInPod(namespace, vertxDeploymentName, "istio-proxy", nil)
+	require.True(t, exists)
+
+	// Confirm that we've created some traces
+	ports := []string{"0:16686"}
+	portForward, closeChan := CreatePortForward(namespace, jaegerInstanceName, "all-in-one", ports, fw.KubeConfig)
+	defer portForward.Close()
+	defer close(closeChan)
+	forwardedPorts, err := portForward.GetPorts()
+	require.NoError(t, err)
+	queryPort := strconv.Itoa(int(forwardedPorts[0].Local))
+
+	url := "http://localhost:" + queryPort + "/api/traces?service=order"
+	err = WaitAndPollForHTTPResponse(url, func(response *http.Response) (bool, error) {
+		body, err := ioutil.ReadAll(response.Body)
+		require.NoError(t, err)
+
+		resp := &resp{}
+		err = json.Unmarshal(body, &resp)
+		require.NoError(t, err)
+
+		return len(resp.Data) > 0 && strings.Contains(string(body), "traceID"), nil
+	})
+	require.NoError(t, err, "SmokeTest failed")
+}
+
+func getBusinessAppCR(err error) *os.File {
```

Comment:
@lujiajing1126 Looks good, just one last thing.  This function looks to have been copied from examples1_test.go.  Can you move it to utils.go and update both tests to access it from there?

Reply:
Done